### PR TITLE
Renovate/python telegram bot 20.x

### PR DESCRIPTION
**Pull request by markusressel**
This PR migrates the existing code according to the official python-telegram-bot docs.

**In addition** to this, this PR allows the omission of the github token, as well as the youtube token.
I mainly did this to be able to test things more easily, but I reckon it can be useful in general.

### CODE REVIEW

File: satisfactory/chat.py [hunk 1]
```diff
@@ -2,79 +2,85 @@
...
+    async def check_videos_timer(self, context: CallbackContext):
+        await self.check_videos(context)
 
     @Summary("satisfactory_bot_send_random_quote_timer", "Time spent running the send_random_quote_timer").time()
-    def send_random_quote_timer(self, context: CallbackContext):
-        self.send_random_quote(context)
-        self.updater.job_queue.run_once(self.send_random_quote_timer, random.randrange(60*60*24, 60*60*24*7, 1))
+    async def send_random_quote_timer(self, context: CallbackContext):
+        await self.send_random_quote(context)
+        (self._app.job_queue.run_once(self.send_random_quote_timer,
```
Comment (ekeih) [inline satisfactory/chat.py:79]: Is the outer `(...)` needed for something?

Comment (markusressel) [reply satisfactory/chat.py:79]: hmm, I didn't add that explictly, so maybe formatting, although I don't think it should be necessary :thinking:

Comment (ekeih) [reply satisfactory/chat.py:79]: 😄 Yeah, looks a bit weird, but it shouldn't hurt, so 🤷

**Comment by ekeih** [comment]
Awesome, thanks! 🙏

**Pull request merged**